### PR TITLE
Let VersionStore.create()+commit() return the current hash

### PR DESCRIPTION
**Pull request by snazy**
Returns the reference created by `TreeApi.createReference()`, useful when creating a branch without
specifying a commit-hash.

Returns the updated `Branch` with the new commit-hash in a new `CommitMultipleOperationsResponse`
object from `TreeApi.commitMultipleOperations()`.

Adopted some usages of both `commitMultipleOperations()` and `createReference()` in the code base.

Moved the `@Deprecated` annotations from `ClientContentsApi` to `ContentsApi`.

Updated the REST examples and updated some related wording issues in comments.

REST API change: both functions in `TreeApi` now return HTTP/200 instead of HTTP/204 (no content).

---
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/1089)

### CODE REVIEW

File: servers/quarkus-server/src/test/java/org/projectnessie/server/TestRest.java [hunk 1]
```diff
@@ -119,10 +121,39 @@ void closeClient() {
   }
 
   @Test
-  void illegalArgumentException() {
-    assertThat(
-        assertThrows(NessieBadRequestException.class, () -> tree.createReference(Hash.of("12341234cafebeef"))).getMessage(),
-        startsWith("Bad Request (HTTP/400): Only tag and branch references can be created"));
+  void createReferences() throws NessieNotFoundException {
```
Comment (rymurr) [inline servers/quarkus-server/src/test/java/org/projectnessie/server/TestRest.java:124]: is it a bad diff render on github or has the `illegalArgumentException` test been removed. If yes then how come?

File: model/src/main/java/org/projectnessie/api/TreeApi.java [hunk 2]
```diff
@@ -407,14 +409,17 @@ void mergeRefIntoBranch(
...
   @APIResponses({
-      @APIResponse(responseCode = "204", description = "Updated successfully."),
+      @APIResponse(responseCode = "200", description = "Updated successfully.",
+        content = {@Content(examples = {@ExampleObject(ref = "commitMultipleOperationsResponse")})}),
       @APIResponse(responseCode = "400", description = "Invalid input, ref/hash name not valid"),
       @APIResponse(responseCode = "404", description = "Provided ref doesn't exists"),
       @APIResponse(responseCode = "409", description = "Update conflict")
   })
-  public void commitMultipleOperations(
+  CommitMultipleOperationsResponse commitMultipleOperations(
```
[the remark this answers is not in the document]
Comment (snazy) [reply model/src/main/java/org/projectnessie/api/TreeApi.java:422]: Was thinking a lot about whether to just return `Reference` or a specific object and went with the latter, so we could return more information like the computed commit-metadata, generated IDs and such things.

File: servers/quarkus-server/src/test/java/org/projectnessie/server/TestRest.java (hunk 1 above)
Comment (snazy) [reply servers/quarkus-server/src/test/java/org/projectnessie/server/TestRest.java:124]: Not gone, just integrated in that new test method.

File: python/pynessie/nessie_client.py [hunk 3]
```diff
@@ -138,10 +143,11 @@ def get_values(self: "NessieClient", ref: str, *tables: str) -> Generator[Conten
         """
         return (ContentsSchema().load(get_table(self._base_url, ref, _format_key(i), self._ssl_verify)) for i in tables)
 
-    def commit(self: "NessieClient", branch: str, old_hash: str, reason: Optional[str] = None, *ops: Operation) -> None:
+    def commit(self: "NessieClient", branch: str, old_hash: str, reason: Optional[str] = None, *ops: Operation) -> dict:
```
[the remark this answers is not in the document]
Comment (snazy) [reply python/pynessie/nessie_client.py]: Oops, yea, wrong return type

File: model/src/main/java/org/projectnessie/api/TreeApi.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (rymurr) [reply model/src/main/java/org/projectnessie/api/TreeApi.java:422]: I am inclined to just use Reference. YAGNI and all that

**Review by rymurr — COMMENTED** [review]
all looks good except for that new object. I sort of feel like we aren't going to need it.

**Comment by snazy** [comment]
> all looks good except for that new object. I sort of feel like we aren't going to need it.

Fair enough.

**Review by rymurr — APPROVED** [review]

**Pull request merged**